pkg/manager/member/tikv_scaler.go

```diff
@@ -191,6 +191,9 @@ func (s *tikvScaler) scaleInOne(tc *v1alpha1.TidbCluster, skipPreCheck bool, upT
 	for _, store := range tc.Status.TiKV.Stores {
 		if store.PodName == podName {
 			state := store.State
+			if pod.Labels[label.StoreIDLabelKey] == "" {
+				return deletedUpStore, fmt.Errorf("StoreID not yet updated on pod label")
+			}
```

Comment:
I'm not sure if there is some edge case where store id in label won't match store id in tc status. If that happens I don't want to block it, as tc status value is more authoritative. The label is only required because we need to confirm tombstone store afterwards. (tc.status won't have store id once deleted). This is for the case 3 situation below.
So I feel better safe to not block deletion if store mismatches somehow (we are deleting & fixing. here anyways)